demos/http3/ossl-nghttp3-demo-server.c

```diff
@@ -0,0 +1,635 @@
+
+/* This callback validates and negotiates the desired ALPN on the server side.
+ */
+static int select_alpn(SSL *ssl, const unsigned char **out,
+                       unsigned char *out_len, const unsigned char *in,
+                       unsigned int in_len, void *arg) {
+  if (SSL_select_next_proto((unsigned char **)out, out_len, alpn_ossltest,
+                            sizeof(alpn_ossltest), in,
+                            in_len) != OPENSSL_NPN_NEGOTIATED)
+    return SSL_TLSEXT_ERR_ALERT_FATAL;
+
+  return SSL_TLSEXT_ERR_OK;
+}
+
+/* Create SSL_CTX. */
+static SSL_CTX *create_ctx(const char *cert_path, const char *key_path) {
+  SSL_CTX *ctx;
+
+  ctx = SSL_CTX_new(OSSL_QUIC_server_method());
+  if (ctx == NULL)
+    goto err;
+
+  /* Load certificate and corresponding private key. */
+  if (SSL_CTX_use_certificate_chain_file(ctx, cert_path) <= 0) {
+    fprintf(stderr, "couldn't load certificate file: %s\n", cert_path);
+    goto err;
+  }
+
+  if (SSL_CTX_use_PrivateKey_file(ctx, key_path, SSL_FILETYPE_PEM) <= 0) {
+    fprintf(stderr, "couldn't load key file: %s\n", key_path);
+    goto err;
+  }
+
+  if (!SSL_CTX_check_private_key(ctx)) {
+    fprintf(stderr, "private key check failed\n");
+    goto err;
+  }
+
+  /* Setup ALPN negotiation callback. */
+  SSL_CTX_set_alpn_select_cb(ctx, select_alpn, NULL);
+  return ctx;
+
+err:
+  SSL_CTX_free(ctx);
+  return NULL;
+}
+
+/* Create UDP socket using given port. */
+static int create_socket(uint16_t port) {
+  int fd = -1;
+  struct sockaddr_in sa = {0};
+
+  if ((fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP)) < 0) {
+    fprintf(stderr, "cannot create socket");
+    goto err;
+  }
+
+  sa.sin_family = AF_INET;
+  sa.sin_port = htons(port);
+
+  if (bind(fd, (const struct sockaddr *)&sa, sizeof(sa)) < 0) {
+    fprintf(stderr, "cannot bind to %u\n", port);
+    goto err;
+  }
+
+  return fd;
+
+err:
+  if (fd >= 0)
+    BIO_closesocket(fd);
+
+  return -1;
+}
+
+static int waitsocket(int fd) {
+  fd_set read_fds;
+  FD_ZERO(&read_fds);
+  FD_SET(fd, &read_fds);
+  int fdmax = fd;
+  if (select(fdmax + 1, &read_fds, NULL, NULL, NULL) == -1) {
+    printf("waitsocket failed\n");
+    exit(1);
+  }
+  printf("waitsocket %d\n", FD_ISSET(fd, &read_fds));
+  return 0;
+}
+
+/* Main loop for server to accept QUIC connections. */
+static int run_quic_server(SSL_CTX *ctx, int fd) {
+  int ok = 0;
+  SSL *listener = NULL, *conn = NULL;
+
+  /* Create a new QUIC listener. */
+  if ((listener = SSL_new_listener(ctx, 0)) == NULL)
+    goto err;
+
+  /* Provide the listener with our UDP socket. */
+  if (!SSL_set_fd(listener, fd))
+    goto err;
+
+  /* Begin listening. */
+  if (!SSL_listen(listener))
+    goto err;
+
+  /*
+   * Listeners, and other QUIC objects, default to operating in blocking mode,
+   * so the below call is not actually necessary. The configured behaviour is
+   * inherited by child objects.
+   */
+  SSL_set_blocking_mode(listener, 1);
```

Comment:
This is declared as _owur (warn on unused result).  You have to check the return code here and handle it

Reply:
fixed